cme/connection.py

```diff
@@ -175,88 +175,86 @@ def login(self):
-                                                if self.plaintext_login(self.domain, usr.strip(), password): return True
-                                            else:
-                                                if self.plaintext_login(usr.strip(), password): return True
-
-                                    elif not isinstance(password, str) and isfile(password.name) and self.args.no_bruteforce == False:
-                                        for f_pass in password:
+                                                if self.hash_login(self.domain, usr.strip(), ntlm_hash_file): return True
+
+                            elif self.args.password:
+                                with sem:
+                                    for password in self.args.password:
+                                        if isfile(password):
+                                            with open(password, 'r') as password_file:
+                                                if self.args.no_bruteforce == False:
+                                                    for f_pass in password_file:
+                                                        if not self.over_fail_limit(usr.strip()):
+                                                            if hasattr(self.args, 'domain'):
+                                                                if self.plaintext_login(self.domain, usr.strip(), f_pass.strip()): return True
+                                                            else:
+                                                                if self.plaintext_login(usr.strip(), f_pass.strip()): return True
+                                                elif self.args.no_bruteforce == True:
+                                                    user_file.seek(0) # HACK: this should really not be in the usr for loop
+                                                    for usr, f_pass in zip(user_file, password_file):
+                                                        if not self.over_fail_limit(usr.strip()):
+                                                            if hasattr(self.args, 'domain'):
+                                                                if self.plaintext_login(self.domain, usr.strip(), f_pass.strip()): return True
+                                                            else:
+                                                                if self.plaintext_login(usr.strip(), f_pass.strip()): return True
+                                        else: # password is a string
                                             if not self.over_fail_limit(usr.strip()):
                                                 if hasattr(self.args, 'domain'):
-                                                    if self.plaintext_login(self.domain, usr.strip(), f_pass.strip()): return True
+                                                    if self.plaintext_login(self.domain, usr.strip(), password): return True
```

Comment:
good catch

Reply:
This is just an artifact of the diff algorithm, this wasn't broken. Seems like `--diff-algorithm=histogram` does a little better, but I think for this change you need to go side-by-side diff as I needed to flip around the code paths for string vs file